### PR TITLE
[Model] Add encoder CUDA graph support to Lfm2VL

### PR DESCRIPTION
**Pull request by vincentzed**
## Purpose
@Liquid4All 's internal commit cut.
LFM2 Moe VL:
Has `SupportsEncoderCudaGraph`.
Bench latency:
~10-20% lower e2e latency for low bs.

## Test Plan
**Accuracy was tested with internal scripts (i.e logits + rouge)**

`MODEL=/root/.cache/huggingface/hub/models--LiquidAI--LFM2-VL-3B/snapshots/97a81f5c18b73352134099952a4856e2553d42d7`

```
vllm bench throughput \
  --model "$MODEL" --served-model-name LiquidAI/LFM2-VL-3B --trust-remote-code \
  --dataset-name custom_image --dataset-path eval/custom_vlm_30k_5img_400.jsonl \
  --backend vllm-chat --output-len 500 --num-prompts 400 \
  --limit-mm-per-prompt '{"image": 1}' \
  --allowed-local-media-path /vllm-workspace/wayfair-data/eval/synth_imgs \
  --mm-processor-cache-gb 0 --no-enable-prefix-caching \
  --max-num-seqs 2048 --max-model-len 32768 --gpu-memory-utilization 0.90 \
  --output-json eval/logs/custom_vlm_3b_30k_400.json
```

Encoder CG flag
```
 --compilation-config '{"cudagraph_mm_encoder": true, "encoder_cudagraph_token_budgets":[64,256,512,1024,2048,4096], "encoder_cudagraph_max_vision_items_per_batch":16}'
```

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://docs.vllm.ai/en/latest/contributing>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm/model_executor/models/lfm2_vl.py [hunk 1]
```diff
@@ -819,6 +803,389 @@ def embed_multimodal(self, **kwargs: object) -> MultiModalEmbeddings:
...
+            padding_logics={
+                "cu_seqlens": _pad_cumulative_seqlens_buffer,
+            },
+        )
+
+    def get_input_modality(
+        self,
+        mm_kwargs: dict[str, Any],
+    ) -> str:
+        return "image"
```
Comment (Isotr0py) [inline vllm/model_executor/models/lfm2_vl.py]:
```suggestion
```
This is same as the default implementation:
https://github.com/vllm-project/vllm/blob/4128605ad4197551a5ae4de08ede38f2348d1eb5/vllm/model_executor/models/interfaces.py#L1523-L1528

File: vllm/model_executor/models/lfm2_vl.py [hunk 2]
```diff
@@ -579,14 +604,24 @@ def forward(
...
 
     hf_to_vllm_mapper = WeightsMapper(
         orig_to_new_prefix={
             "lm_head.": "language_model.lm_head.",
             "model.language_model.": "language_model.model.",
+            "model.vision_tower.embeddings.": "vision_tower.vision_model.embeddings.",
+            "model.vision_tower.encoder.": "vision_tower.vision_model.encoder.",
+            "model.vision_tower.post_layernorm.": (
+                "vision_tower.vision_model.post_layernorm."
+            ),
```
Comment (Isotr0py) [inline vllm/model_executor/models/lfm2_vl.py]: Can we just replace `"model.vision_tower.": "vision_tower."` below with `"model.vision_tower.": "vision_tower.vision_model."`?

File: vllm/model_executor/models/lfm2_vl.py [hunk 3]
```diff
@@ -712,69 +748,17 @@ def image_pixels_to_features(
...
-                    f"got ({height}, {width}) with factor={factor}."
-                )
-            projected_lengths_list.append((height // factor) * (width // factor))
-
-        projected_packed = self.multi_modal_projector(
-            vision_features_packed=vision_features_packed,
-            spatial_shapes=spatial_shapes,
+        projected_packed = self.encoder_cudagraph_forward(values)
+        projected_lengths_list = self._get_lfm2vl_tile_output_lengths(
+            spatial_shapes_list
```
Comment (Isotr0py) [inline vllm/model_executor/models/lfm2_vl.py:798]: Encoder CG manager should be able to execute graph capture/replay automatically, so there is no need to modify existing eager forwarding method:
https://github.com/vllm-project/vllm/blob/4128605ad4197551a5ae4de08ede38f2348d1eb5/vllm/v1/worker/gpu_model_runner.py#L3058-L3073

File: vllm/model_executor/models/lfm2_vl.py (hunk 1 above)
Comment (vincentzed) [reply vllm/model_executor/models/lfm2_vl.py]: thank you, fixed in b892185!

File: vllm/model_executor/models/lfm2_vl.py (hunk 2 above)
Comment (vincentzed) [reply vllm/model_executor/models/lfm2_vl.py]: thank you, fixed in 32ca211! (the LFM2-VL/LFM2.5-VL checkpoints store these as `model.vision_tower.vision_model.*`, so the existing catch-all already covered them — dropped the extra mappings instead)

File: vllm/model_executor/models/lfm2_vl.py (hunk 3 above)
Comment (vincentzed) [reply vllm/model_executor/models/lfm2_vl.py:798]: thank you, fixed in f60b8fd!

**Comment by vincentzed** [comment]
@Isotr0py we tested mmmu and the acc is still good. Let me know if any other changes

File: vllm/model_executor/models/lfm2_vl.py [hunk 4]
```diff
@@ -819,6 +840,383 @@ def embed_multimodal(self, **kwargs: object) -> MultiModalEmbeddings:
...
+                "gather_idx",
+            ],
+            out_hidden_size=self.config.text_config.hidden_size,
+            padding_logics={
+                "cu_seqlens": _pad_cumulative_seqlens_buffer,
+            },
+        )
+
+    def get_max_frames_per_video(self) -> int:
+        return 0
```
Comment (Isotr0py) [inline vllm/model_executor/models/lfm2_vl.py:864]:
```suggestion
```
Seems unused.

**Review by Isotr0py — APPROVED** [review]
